### DISCUSSION TITLE
Search dialog with per-track indexing?

### DISCUSSION THREAD

**Discussion by bbimber** [GMOD/jbrowse-components#2339] [General]
I have a jbrowse 2 session where my config contains a GFF track. This track was indexed standalone using "jbrowse text-search --file xxxx.gff". When I type into the search box i get the expected typeahead. When I pick a feature and the dialog pops up, i get:

![image](https://user-images.githubusercontent.com/7724863/133691981-3dff96d6-2e82-4d25-971c-9e22fbd35213.png)

The results are valid, but is there a reason track is "N/A"? I'm guessing maybe this has something to do with aggregate indexes (which need to be aware of track), and per-track indexes? The code must "know" the track for these results though?

**Comment by cmdcolin** [comment]
We will probably want to clarify the docs for this. I believe you can use the filename that you passed to --file as the trackId e.g. 

If you use --file volvox.sort.gff3.gz

Then make the trackId on this track volvox.sort.gff3.gz

**Reply by bbimber** [reply]
But doesnt the config JSON inherently connect track to search config, at least when using per-track indexing? If a track is potentially used across more than one session (presumably the point of per-track indexing), is there any reason trackId needs to be defined at time of indexing?

**Reply by cmdcolin** [reply]
conceptually that is true, but internally, our code treats per track indexes the same as aggregate indexes, so it expects that the track id on the track to match what is in the index, and for --file, it autoassigns a trackid to the index

we could possibly reconsider this but it is how the ship sailed in the v1.4.4 release

**Reply by bbimber** [reply]
OK, this is not exactly a deal-breaker.

Question then: could we place a trackId property on the searchConfig object? If present, could the code just assume 100% of results from this adapter are from this trackId?

**Reply by cmdcolin** [reply]
I could look into this possibility...it does seem like it would be intuitive. the text-index tool could also output a little config snippet to recommend to the user also

**Reply by bbimber** [reply]
I dont know the guts of jbrowse well enough, but does whatever code parses the search config object have access to the owning track? if the search config is a child of a track (as opposed to an actual aggregate search index), could code automatically inject a "defaultTrackId" property into it? This would avoid a person ever needing to specify that? And then the code that retrieves search results would either take the trackID of the search result (if provided), or fall back to the defaultTrackId property (if specified)? Whatever the precise implementation, this just seems like something the code ought to be able to translate

**Reply by cmdcolin** [reply]
I created a issue in #2344 to try to keep track of this...would have to dig into it a lil bit to see
